app/renderer/js/components/tab.ts

```diff
@@ -1,44 +1,44 @@
 'use strict';
 
-const BaseComponent = require(__dirname + '/../components/base.js');
+import WebView = require('./webview');
+import BaseComponent = require('./base');
+
+// TODO: TypeScript - Type annotate props
+interface TabProps {
+	[key: string]: any;
+}
 
 class Tab extends BaseComponent {
-	constructor(props) {
+	props: TabProps;
+	webview: WebView;
+	$el: Element;
+	constructor(props: TabProps) {
 		super();
 
 		this.props = props;
 		this.webview = this.props.webview;
-
-		this.init();
-	}
-
```

Comment:
@vsvipul, @priyank-p can you guys explain why we have removed this?

Reply:
this.template() used in init() function was a property of its child classes - ServerTab and FunctionalTab . So this init() function was moved to its both the subclasses.

Reply:
See commit message of this [commit](https://github.com/zulip/zulip-desktop/pull/782/commits/340023e498e8765981c02447b10a4f8bee359228) to see the relevant changes also.